src/cpu/flexcpu/simple_dataflow_thread.cc

```diff
@@ -158,7 +169,96 @@ SDCPUThread::advanceInst()
     inflightInsts.push_back(dynamic_inst_ptr);
 
     fetchOffset = 0;
-    onBeginFetch(weak_ptr<InflightInst>(dynamic_inst_ptr));
+    attemptFetch(weak_ptr<InflightInst>(dynamic_inst_ptr));
+}
+
+void
+SDCPUThread::attemptFetch(weak_ptr<InflightInst> inst)
```

Comment:
I don't like this name...

Fetching is when you get data from memory. Decoding is when you decode... This is moving the instruction that has been fetched into the decoder. I'm not sure what the name *should* be. 

Why is this a separate step? Why doesn't the decoder just get data directly from the fetch buffer?

Reply:
This was just the onBeginFetch function renamed. When I made this feature change, I tried to do it in a way that minimized the change from the flow logic before. Since the buffer is functionally like an internal icache, it now includes logic to check that "cache" before sending a response to the external icache.